forum/qa-content/qa-page.js

```diff
@@ -177,6 +177,58 @@ function qa_ajax_error()
 }
 
 
+/**
+ *	Feature: check if browser supports copying to clipboard.
+ *	If true, return function, which can copy highlighted content to clipboard on button click - then user can paste it wherever he wants into.
+ *	Otherwise return null.
+ */
+window.initializeCopyToClipboard = ( function() {
+	// check if browser supports 'select()' and 'copy' commands
+	var isClipboardSupported = ( window.getSelection && document.queryCommandSupported( 'copy' ) && navigator.userAgent.indexOf( 'Firefox' ) === -1 );
+
+	function copyToClipBoard( ev, contentToCopy ) {
+		// prevent page refresh (or something weird) as default button action
+		ev.preventDefault();
+
+		if ( !ev.defaultPrevented ) {
+			return false;
+		}
+
+		/*
+		 * In order to be able to copy the code inside block into the clipboard - so user can easily paste it wherever he wants - within single button click
+		 * a code must be first selected (or highlighted in human meaning), so JavaScript can copy it.
+		 * However selecting is only possible on HTML elements that are 'inputs', such as <textarea>.
+		 * That's why below code creates <textarea> (which has 'display: none'; in CSS) and inserts there code content from block,
+		 * copy it to clipboard and removes it (so user can't really see temporary <textarea> element appears) after whole process.
+		 */
+		var textArea = document.createElement( "textarea" );
+		textArea.classList.add( 'content-copy' );
+		textArea.value = contentToCopy;
+
+		document.body.appendChild( textArea );
+
+		// if anything on the page is selected (a.k.a highlighted) - clear the selection
+		if ( window.getSelection().rangeCount )
+			window.getSelection().removeAllRanges();
+
+		/*
+		 * Below code will select given DOM elements
+		 * and create Range Object, so that text content can be selected (a.k.a highlighted)
+		 * Modified script from source: http://stackoverflow.com/a/1173319/4983840
+		 */
+		var range = document.createRange();
+		range.selectNode( textArea );
+		window.getSelection().addRange( range );
+
+		// copy content that is select inside Document - so that is only textarea
+		document.execCommand( 'copy' );
+
+		// remove <textarea> from DOM
+		document.body.removeChild( textArea );

```

Comment:
Again, stating obvious. Don't write redundant comments. Every dev should understand that `body.removeChild( textArea );` is removing textArea from body. The code never lies, comments sometimes does. 
```
// iterate over array
for (int i=0; i<size; i++) { array[i]; (...)}
```